xdsl/dialects/builtin.py

```diff
@@ -1388,11 +1388,25 @@ def from_params(handle: str | StringAttr, type: Attribute) -> DenseResourceAttr:
         return DenseResourceAttr([handle, type])
 
 
+DenseArrayT = TypeVar(
+    "DenseArrayT",
+    bound=IntegerType | AnyFloat,
+    default=IntegerType | AnyFloat,
+    covariant=True,
+)
+_IntegerTypeInvT = TypeVar("_IntegerTypeInvT", bound=IntegerType, default=IntegerType)
```

Comment:
is this the best place for this?

Reply:
I'm not sure tbh, we don't really seem to have a naming scheme or place for type vars but I'll move it to just after `IntegerType`